plugins/modules/swim_workflow_manager.py

```diff
@@ -1064,119 +1255,158 @@ def get_diff_import(self):
+                    import_params = dict(
+                        is_third_party=self.want.get("local_import_details").get("is_third_party"),
+                        third_party_vendor=self.want.get("local_import_details").get("third_party_vendor"),
+                        third_party_image_family=self.want.get("local_import_details").get("third_party_image_family"),
+                        third_party_application_type=self.want.get("local_import_details").get("third_party_application_type"),
+                        multipart_fields={'file': (os.path.basename(file_path), open(file_path, 'rb'), 'application/octet-stream')},
+                        multipart_monitor_callback=None
+                    )
+                    import_function = 'import_local_software_image'
+
+                else:  # CCO import
+                    image_name = images_to_import[0]
+                    cco_image_id = self.get_cco_image_id(image_name)
+                    import_params = {"id": cco_image_id}
+                    import_function = 'download_the_software_image'
+
+                self.log("importing with the import_params - {0}".format(import_params))
+                response = self.dnac._exec(
+                    family="software_image_management_swim",
+                    function=import_function,
+                    op_modifies=True,
+                    params=import_params,
                 )
-                import_function = 'import_local_software_image'
-
-            response = self.dnac._exec(
-                family="software_image_management_swim",
-                function=import_function,
-                op_modifies=True,
-                params=import_params,
-            )
-            self.log("Received API response from {0}: {1}".format(import_function, str(response)), "DEBUG")
+                self.log("Received API response from {0}: {1}".format(import_function, str(response)), "DEBUG")
 
-            task_details = {}
-            task_id = response.get("response").get("taskId")
-
-            while (True):
-                task_details = self.get_task_details(task_id)
-                name = image_name.split('/')[-1]
+                task_details = {}
+                task_id = response.get("response").get("taskId")
 
-                if task_details and \
-                        ("completed successfully" in task_details.get("progress").lower()):
-                    self.result['changed'] = True
-                    self.status = "success"
-                    self.msg = "Swim Image {0} imported successfully".format(name)
-                    self.result['msg'] = self.msg
-                    self.log(self.msg, "INFO")
-                    break
+                # Monitor the task progress
+                while True:
+                    task_details = self.get_task_details(task_id)
+                    if task_details and "completed successfully" in task_details.get("progress", "").lower():
```

Comment:
add a new line before this if check..

Instead of checking task_details every time, can we optimize the code?
Get the progress and store it.


```
while True:
    task_details = self.get_task_details(task_id)
    
    if not task_details:
        self.log("Failed to retrieve task details.", "ERROR")
        self.status = "failed"
        self.msg = "Unable to retrieve task details for task ID {0}.".format(task_id)
        self.result['response'] = self.msg
        return self

    progress = task_details.get("progress", "").lower()
    if "completed successfully" in progress:
        if images_to_import:
                images_to_import_str = ", ".join(images_to_import)
                self.result['changed'] = True
                self.status = "success"
                self.msg = "SWIM Image(s) {0} imported successfully.".format(images_to_import_str)
                self.result['msg'] = self.msg
                self.log(self.msg, "INFO")
                break

     if task_details.get("isError"):
            failure_reason = task_details.get("failureReason", "").lower()
            if "already exists" in failure_reason:
                self.msg = "SWIM Image(s) {0} already exists in Cisco Catalyst Center.".format(", ".join(images_to_import))
                self.result['msg'] = self.msg
                self.status = "success"
                self.result['changed'] = False
                break
            
            self.status = "failed"
            self.msg = task_details.get("failureReason", "SWIM Image(s) {0} seems to be invalid.".format(", ".join(images_to_import)))
            self.result['response'] = self.msg
            self.log(self.msg, "WARNING")
            return self
```

Reply:
but for getting the task details we need to run in loop. if we want we can reduce the times task detail is called during the check